src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/JsonObject.cs

```diff
@@ -377,13 +396,27 @@ private static Hashtable PopulateHashTableFromJDictionary(JObject entries, out E
                     case JObject dic:
                         {
                             // Dictionary
-                            var dicResult = PopulateHashTableFromJDictionary(dic, out error);
+                            var dicResult = PopulateHashtableFromJDictionary(dic, ordered, out error);
                             if (error != null)
                             {
                                 return null;
                             }
 
-                            result.Add(entry.Key, dicResult);
+                            if (!Ordered)
+                            {
+                                Hashtable htResult = new(dicResult.Count);
```

Comment:
This 'fill' code snippet is used three times, so can be made into a simple static util function.